### ISSUE TITLE
Jest is broken

### ISSUE THREAD

**Issue by kachnitel** [kachnitel/RideTime#19]
[jest.log](https://github.com/kachnitel/RideTime/files/2947596/jest.log)

```
Jest encountered an unexpected token
      This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.
```

**Comment by kachnitel** [comment]
5e3ce5995c22da05e95f3da3113caca45a05d60f allows Jest to run, Mocks still not working:

`TypeError: Cannot read property 'default' of undefined`

[jest.log](https://github.com/kachnitel/RideTime/files/2947951/jest.log)

**Comment by kachnitel** [comment]
```
 FAIL  stores/__tests__/UserStore.mobx-test.js
  ● Test suite failed to run

    TypeError: _RidersProvider.default is not a constructor

      19 | 
      20 | rootStore.appStore = applicationStore
    > 21 | rootStore.userStore = new UserStore(rootStore, new RidersProvider())
         |                                                         ^
      22 | export default rootStore
      23 | 

      at Object.<anonymous> (stores/RootStore.singleton.js:21:57)
      at Object.<anonymous> (src/Connection.js:6:41)
      at Object.<anonymous> (providers/RidersProvider.js:1:327)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.963s
Ran all test suites related to changed files.
  console.log stores/RootStore.singleton.js:17
    Hydrated persisted data  ApplicationStore {
      rootStore: RootStore { appStore: [Circular] },
      _userId: [Getter/Setter],
      _accessToken: [Getter/Setter] }

```

Another fail in a file that doesn't cause issues in app, only in test. Maybe the Provider should be mocked but even that just doesn't seem to work.

**Comment by kachnitel** [comment]
Debugging it appears in `node_modules/@jest/core/node_modules/jest-haste-map/build/ModuleMap.js` in `getModule` that module is null and is trying to mock. 
Call stack:
```
requireModuleOrMock (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:472)
Object.<anonymous> (/home/kachna/workspace/projects/RideTime/src/Connection.js:6)
_execModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:765)
requireModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:361)
requireModuleOrMock (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:469)
Object.<anonymous> (/home/kachna/workspace/projects/RideTime/providers/RidersProvider.js:1)
_execModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:765)
requireModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:361)
requireModuleOrMock (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:469)
Object.<anonymous> (/home/kachna/workspace/projects/RideTime/stores/UserStore.mobx.js:2)
_execModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:765)
requireModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:361)
requireModuleOrMock (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:469)
Object.<anonymous> (/home/kachna/workspace/projects/RideTime/stores/__tests__/UserStore.mobx-test.js:1)
_execModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:765)
requireModule (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-runtime/build/index.js:361)
(anonymous function) (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-jasmine2/build/index.js:202)
asyncGeneratorStep (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-jasmine2/build/index.js:27)
_next (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-jasmine2/build/index.js:47)
(anonymous function) (/home/kachna/workspace/projects/RideTime/node_modules/@jest/core/node_modules/jest-jasmine2/build/index.js:52)
```

**Comment by kachnitel** [comment]
[SO](https://stackoverflow.com/questions/55311458/type-hinting-vs-circular-reference-in-javascript)
[Reddit](https://www.reddit.com/r/reactnative/comments/b3f6yj/expo_jest_testing_struggle/)

**Comment by kachnitel** [comment]
dc42af4ccf21d759ef0edc305730c2b61a75877b fixes the mock problem, `_RidersProvider.default is not a constructor` is still an issue.

**Comment by kachnitel** [comment]
Fixed in 9cd9068
